src/plugins/record.ts

```diff
@@ -167,6 +174,25 @@ class RecordPlugin extends BasePlugin<RecordPluginEvents, RecordPluginOptions> {
     }
   }
 
+  /** Get a list of available audio devices */
+  public async getAvailableAudioDevices() {
+    // Request access to the microphone before enumerating devices, otherwise
+    // the browser will not return any device IDs
+    let stream: MediaStream
+    try {
+      stream = await navigator.mediaDevices.getUserMedia({
+        audio: true,
+      })
+    } catch (err) {
+      throw new Error('Error getting audio devices: ' + (err as Error).message)
+    }
+    this.stream = stream
+    this.stopMic()
```

Comment:
Looks good overall, thank you!

These lines are almost the same as in `startMic`, should we just call it here? Or maybe throw an error like `Please start the microphone first`.

Reply:
Possibly. I was concerned about how these lines from `startMic` would affect it? It doesn't need to render the stream when calling the function. What do you think?
```
    const onDestroy = this.renderMicStream(stream)
    this.subscriptions.push(this.once('destroy', onDestroy))
```

Reply:
I could also use this: https://developer.mozilla.org/en-US/docs/Web/API/Permissions/query
Then if no permission - call startMic

Reply:
Sorry for all the suggestions here. It would be nice if this was actually a static function and didn't touch the main class properties. It would keep similar code here but just use the local variable and use similar code to `stopMic` to ensure the steam gets deallocated.

Reply:
Landed on removing starting a stream and just adding better docs. 
The permissions api I shared isn't doesn't support a microphone query in Firefox 🤷🏻‍♂️